rocketpy/motors/Motor.py

```diff
@@ -149,88 +164,196 @@ def __init__(
+        self, newBurnTime, totalImpulse, oldTotalImpulse=None, startAtZero=True
     ):
-        """Transforms the thrust curve supplied by changing its total burn time
-        and/or its total impulse, without altering the general shape of the
-        curve. May translate the curve so that thrust starts at time equals 0,
-        without any delays.
+        """Transforms the thrust curve supplied by changing its total
+        burn time and/or its total impulse, without altering the
+        general shape of the curve. May translate the curve so that
+        thrust starts at time equals 0, without any delays.
 
         Parameters
         ----------
-        burnTime : float
-            New desired burn out time in seconds.
+        newBurnTime : float, tuple of float
+            New desired burn time in seconds.
         totalImpulse : float
             New desired total impulse.
         oldTotalImpulse : float, optional
-            Specify the total impulse of the given thrust curve, overriding the
-            value calculated by numerical integration. If left as None, the
-            value calculated by numerical integration will be used in order to
-            reshape the curve.
+            Specify the total impulse of the given thrust curve,
+            overriding the value calculated by numerical integration.
+            If left as None, the value calculated by numerical
+            integration will be used in order to reshape the curve.
         startAtZero: bool, optional
-            If True, trims the initial thrust curve points which are 0 Newtons,
-            translating the thrust curve so that thrust starts at time equals 0.
-            If False, no translation is applied.
+            If True, trims the initial thrust curve points which
+            are 0 Newtons, translating the thrust curve so that
+            thrust starts at time equals 0. If False, no translation
+            is applied.
 
         Returns
         -------
         None
         """
         # Retrieve current thrust curve data points
-        timeArray = self.thrust.source[:, 0]
-        thrustArray = self.thrust.source[:, 1]
+        timeArray, thrustArray = self.thrust.xArray, self.thrust.yArray
+
         # Move start to time = 0
         if startAtZero and timeArray[0] != 0:
             timeArray = timeArray - timeArray[0]
 
-        # Reshape time - set burn time to burnTime
-        self.thrust.source[:, 0] = (burnTime / timeArray[-1]) * timeArray
-        self.burnOutTime = burnTime
-        self.thrust.setInterpolation(self.interpolate)
+        # Reshape time - set burn time to newBurnTime
+        self.burn_time = newBurnTime
 
-        # Reshape thrust - set total impulse
+        # Compute old thrust based on new time discretization
+        newTimeArray = (
+            (self.burn_time[1] - self.burn_time[0]) / (timeArray[-1] - timeArray[0])
+        ) * timeArray + self.burn_time[0]
```

Comment:
I believe the current implementation only works if `timeArray[0] == 0`.

```suggestion
        # Adjust scale
        newTimeArray = (
            (self.burn_time[1] - self.burn_time[0]) / (timeArray[-1] - timeArray[0])
        ) * timeArray
        # Adjust origin
        newTimeArray = newTimeArray - newTimeArray[0] + self.burn_time[0]
```